### DISCUSSION TITLE
Documentation for time.maketime()

### DISCUSSION THREAD

**Discussion by brianreinhold** [micropython/micropython#14034] [General]
I look at the documentation and I see

> time.mktime()[¶](https://docs.micropython.org/en/latest/library/time.html#time.mktime)
This is inverse function of localtime. It’s argument is a full 8-tuple which expresses a time as per localtime. It returns an integer which is the number of seconds since Jan 1, 2000.

Why isn't the parameter placed in the method? When I try and put that Tuple in the method, pylance gives me an error saying 0 positional arguments are expected. Click on it to go to the definition and I see exactly the same thing as in the on-line documentation I quoted above. Of course, pylance can't read comments, it reads code and since there are no parameters it gives me an error.

However, when I run the program with no arguments, I get a `ValueError` saying 1 positional argument is expected. Since running is more important than trying to get rid of the red squiggles indicating an error, I put in the tuple.

This is the only such inconsistency I have found (so far). There must be a reason for that. Anyone know why?

**Comment by scruss** [comment]
This does what I'd expect:
```
MicroPython v1.22.1 on 2024-01-05; ESP module with ESP8266
Type "help()" for more information.
>>> import time
>>> time.localtime() ; time.mktime(time.localtime())
(2024, 3, 5, 16, 37, 55, 1, 65)
762971875
>>> 
```
Does your pylance installation know about the differences between MicroPython and Python?

**Reply by brianreinhold** [reply]
@scruss  Do you know what happens if the day of week and yearday are not entered in the tuple? Taking a common date like 2021/03/05 17:26:11 it is difficult to determine the day of the week and the yearday. These are not commonly provided. I need to take and arbitrary (common) time stamp and turn it to an epoch. I wont know yearday or day of week

**Reply by scruss** [reply]
mktime() doesn't seem to care about the last two entries:

```
>>> time.mktime((2024, 3, 5, 16, 37, 55, 1, 65)) == time.mktime((2024, 3, 5, 16, 37, 55, 0, 0)) == time.mktime((2024, 3, 5, 16, 37, 55, -99, -99)) == time.mktime((2024, 3, 5, 16, 37, 55, None, None))
True
```

Y'know, you've got a REPL right there yourself, it's not difficult to try stuff ...

**Reply by brianreinhold** [reply]
Python is new to me. Never thought about REPL. I have been writing scripts and testing that way

**Reply by scruss** [reply]
Maybe I'm over-reliant on the REPL, but for me, it's where I test code before it goes in the editor. I find it particularly useful for testing new sensors/devices: does this device show up on my I2C bus?, etc

MicroPython's documentation may not be quite what a beginner would expect. Many features are dependent on the port (hardware: ESP8266, ESP32, Raspberry Pi Pico, etc) and sometime the documentation for that port isn't as easy to find (or as complete) as it should be. Please keep asking questions here!

**Reply by Josverl** [reply]
@brianreinhold, I think this is a case where incomplete/incorrect documentation cascades into the micropython-stubs and pylance

The `time` module is indeed notoriously different between ports, and some of its methods even take the same arguments in different orders.
I think several doc PRs are stalled in queue due to that.
Perhaps more importantly, the differences between the ports are being addressed, but that's a gradual change.

So for now this calls for a `type: ignore`

**Reply by samneggs** [reply]
I just spent that last couple hours fighting bad documentation for the Pico machine RTC:
rtc.datetime((2017, 8, 23, 2, 12, 48, 0, 0)) # set a specific date and time, eg. 2017/8/23 1:12:48

The general machine.RTC documentation is better. (not Pico specific)

**Reply by brianreinhold** [reply]
yes, the RTD date time on hte PICO has the day of the week in the middle.  The gmtime and localtime have it near at the end with the yearday. . Fortunately, you dont need to know those parameters to set the time.

**Reply by robert-hh** [reply]
See issue  #10578 and the related PR.
